File: java/src/televentas/Pago.java
package televentas;

import java.time.LocalDate;

public abstract class Pago {

    protected double monto;
    protected LocalDate fechaPago;
    protected String estado;

    protected Pago(double monto, LocalDate fechaPago, String estado) {
        this.monto = monto;
        this.fechaPago = fechaPago;
        this.estado = estado;
    }

    protected Pago(double monto, LocalDate fechaPago) {
        this(monto, fechaPago, "pendiente");
    }

    public abstract boolean procesar();

    public abstract boolean validar();

    public double getMonto() {
        return monto;
    }

    public LocalDate getFechaPago() {
        return fechaPago;
    }

    public String getEstado() {
        return estado;
    }
}

File: javascript/src/televentas/pago.js
export class Pago {
  /**
   * @param {number} monto
   * @param {Date} fechaPago
   * @param {string | undefined} estado
   */
  constructor(monto, fechaPago, estado) {
    if (new.target === Pago) {
      throw new TypeError("No se puede instanciar la clase abstracta Pago");
    }
    this.monto = monto;
    this.fechaPago = fechaPago;
    this.estado = estado ?? "pendiente";
  }

  getMonto() {
    return this.monto;
  }

  getFechaPago() {
    return this.fechaPago;
  }

  getEstado() {
    return this.estado;
  }

  procesar() {
    throw new TypeError("Metodo abstracto");
  }

  validar() {
    throw new TypeError("Metodo abstracto");
  }
}
